ompi/mpi/c/init_thread.c.in

```diff
@@ -39,22 +40,32 @@
 PROTOTYPE ERROR_CLASS init_thread(INT_OUT argc, ARGV argv, INT required,
                                   INT_OUT provided)
 {
-    int err, safe_required = MPI_THREAD_SERIALIZED;
+    int err, err_arg_required = false, safe_required = MPI_THREAD_SERIALIZED;
     char *env;
 
     ompi_hook_base_mpi_init_thread_top(argc, argv, required, provided);
 
     /* Detect an incorrect thread support level, but dont report until we have the minimum
      * infrastructure setup.
      */
-    if( (MPI_THREAD_SINGLE == required) || (MPI_THREAD_SERIALIZED == required) ||
-        (MPI_THREAD_FUNNELED == required) || (MPI_THREAD_MULTIPLE == required) ) {
+    err_arg_required = (required != MPI_THREAD_SINGLE && required != MPI_THREAD_FUNNELED &&
+                        required != MPI_THREAD_SERIALIZED && required != MPI_THREAD_MULTIPLE);
+    if (!err_arg_required) {
+        safe_required = required;
+    }
 
-        if (NULL != (env = getenv("OMPI_MPI_THREAD_LEVEL")))  {
-            safe_required = atoi(env);
-        }
-        else {
-            safe_required = required;
+    /* check for environment overrides for required thread level.  If
+     * there is, check to see that it is a valid/supported thread level.
+     *  if it is lower than argument required, set it anyway, program can
```

Comment:
The comment says that you are checking for a lower value, but the code is not checking lower vs. higher.

I personally think it's fine to have the env variable *always* override (regardless of lower vs. higher); please adjust the comment.